### DISCUSSION TITLE
How to use getInfiniteQueryData()

### DISCUSSION THREAD

**Discussion by jamezening** [trpc/trpc#2646] [Q&A] [answered]
I am trying to get infinite query data using [getInfiniteQueryData()](https://trpc.io/docs/v9/useInfiniteQuery#getinfinitequerydata) as shown in the docs.

My `listings` useInfiniteQuery below

```
const listings = trpc.useInfiniteQuery(
    [
      'get-listings',
      {
        initialStart: INITIAL_PAGINATION_START,
        initialLimit: INITIAL_PAGINATION_LIMIT,
        paginationStep: PAGINATION_STEP,
      },
    ],
    {
      getNextPageParam: (lastPage, allPages) => {
       // ...
      keepPreviousData: true,
    },
  );
```

is retrieving paginated data fine, but when I use the code:

```
import { trpc } from '../utils/trpc';

export function MyComponent() {
  const utils = trpc.useContext();
  const allListings = utils.getInfiniteQueryData([
   'get-listings',
      {
        initialStart: INITIAL_PAGINATION_START,
        initialLimit: INITIAL_PAGINATION_LIMIT,
        paginationStep: PAGINATION_STEP,
      },
  ]);
}
```

`allListings` is undefined.

Might there be additional [context](https://trpc.io/docs/v9/context) etc. settings that need to be configured to retrieve infinite query data using `utils.getInfiniteQueryData`?

**Comment by danielreuter** [comment]
Did you ever figure this out @vanilla-james

**Reply by jamezening** [reply]
@danielreuter 
I am grokking my code to figure out how I worked this out.

It looks like I ended up passing the data as props, and not fetching them from the nested components using `utils.getInfiniteQueryData`.

Might be an anti-pattern, but it enabled me to use the data in nested props.

If some code snippets might help, let me know and I'll post some post.

**Comment by danielreuter** [comment, marked as answer]
Actually I figured it out @vanilla-james -- there's a special way to configure tRPC to use react-query: https://trpc.io/docs/reactjs/setup

**Reply by jamezening** [reply]
Awesome, thanks for pointing that out.
